scm-ui/ui-extensions/src/binder.ts

```diff
@@ -52,7 +58,22 @@ export class Binder {
    * @param extension provided extension
    * @param predicate to decide if the extension gets rendered for the given props
    */
-  bind(extensionPoint: string, extension: any, predicate?: Predicate, extensionName?: string) {
+  bind<E extends ExtensionPointDefinition<string, unknown, undefined>>(
+    extensionPoint: E["name"],
+    extension: E["type"]
+  ): void;
+  bind<E extends ExtensionPointDefinition<string, unknown, any>>(
+    extensionPoint: E["name"],
+    extension: E["type"],
+    predicate?: Predicate<E["props"]>,
+    extensionName?: string
+  ): void;
+  bind<E extends ExtensionPointDefinition<string, unknown, any>>(
+    extensionPoint: E["name"],
+    extension: E["type"],
+    predicate?: Predicate<E["props"]>,
+    extensionName?: string
+  ) {
```

Comment:
I don't get this. Why is one void and the other actually do things. Where do we need this?